src/ansi.rs

```diff
@@ -1525,6 +1595,15 @@ where
                     }
                 }
             },
+            ('m', [b'>']) => {
+                let mode = match (next_param_or(1) == 4).then(|| next_param_or(0)) {
```

Comment:
Using `next_param_or(1) == 4` seems very odd to me.

Probably better to just do the following:

```
if let Some(&[4]) = params_iter.next() {
    …
}
```

Or some variation of this (there's probably better options that don't indent so heavily).

Reply:
I'm not sure you can do any better than what I use, simply because you'd need mulitple `unhandled!` cases.

Reply:
Nah, I don't like that, it's just bad, because the closure holds the borrow and you'd have to clone/map. We have an example bellow which looks like:

```rust
                let bottom =
                    params_iter.next().map(|param| param[0] as usize).filter(|&param| param != 0);

```

I'd basically need the same just to check that it's `4`.

Reply:
Ah `next_param_or` captures `params_iter` quite unfortunately.